homeassistant/components/loqed/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for loqed integration."""
+from __future__ import annotations
+
+import json
+import logging
+from typing import Any
+
+import aiohttp
+from loqedAPI import loqed
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import webhook
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.const import CONF_HOST, CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any], zeroconf_host: str | None
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    newdata = data
+    json_config = json.loads(data["config"])
+    newdata["ip"] = json_config["bridge_ip"]
+    newdata["host"] = json_config["bridge_mdns_hostname"]
+    newdata["bkey"] = json_config["bridge_key"]
+    newdata["key_id"] = int(json_config["lock_key_local_id"])
+    newdata["api_key"] = json_config["lock_key_key"]
+
+    if zeroconf_host is not None and zeroconf_host != newdata["host"]:
+        raise InvalidAuth(
```

Comment:
The json blob the user has to copy contains the hostname of the lock, if it doesn't match it means the user pasted in the wrong json blob

Reply:
We should never ask a user to pass a JSON blob as input.

Reply:
The lock needs 4 parameters: Api key,  key Index, client secret. And a hostname. Not copying the json blob means copying more data with a higher risk of errors. If you prefer, we could change it, but it would require some more rework in the library

Reply:
Why not ask the user for their username and password and let Home Assistant fetch the data itself?

Asking a user to log in, copy a JSON blob, or other things with names they probably have no idea what they mean isn't really friendly.

Reply:
I really like the username password idea, I'll investigate whether that is possible. Currently using the username password for the webapp will log you out of the native app (as discussed below) so we need to only use them at a time when the user is aware of the effects.

Reply:
Logging in with username / password can also be protected with a 2FA. I wouldn't know how to deal with that.

Reply:
Don't they have OAuth support? That would make everything way easier, allowing to import device data.

Reply:
They have oauth, but they prefer not to use it as it would require them to have the signing key for the lock on their servers  (or so I'm told) which they prefer not to because of security

Reply:
Seems like Oauth is in the works. But still not soonly realised